url43a0865bb0_kevinjdolan_darwin_tgz-pJ8-tui_DriverJ8/src/darwin/work/threadPool/Worker.java

```diff
@@ -0,0 +1,85 @@
+package darwin.work.threadPool;
+
+import java.util.HashMap;
+
+import darwin.work.BusyException;
+import darwin.work.Job;
+
+/**
+ * The worker class contains a single worker thread
+ * @author Kevin Dolan
+ */
+public class Worker extends Thread{
+	
+	private ThreadPool master;
+	private boolean alive, busy;
+	private Job job;
+	private Object parameters;
+	private HashMap<String, Object> shared;
+	
+	/**
+	 * Create a new worker, with a given master
+	 * @param master the threadpool to which this reports
+	 */
+	public Worker(ThreadPool master) {
+		super("Worker");
+		this.master = master;
+		busy = false;
+		alive = true;
+		shared = new HashMap<String, Object>();
+		start();
+	}
+	
+	/**
+	 * Put a new value to the shared data object
+	 * @param name  the name of the field
+	 * @param value the value to put
+	 */
+	public void putSharedData(String name, Object value) {
+		shared.put(name, value);
+	}
+	
+	/**
+	 * @return true if this worker is processing something
+	 */
+	public boolean isBusy() {
+		return busy;
+	}
+	
+	/**
+	 * Start this worker on a job
+	 * @param job 		 the job to be performed
+	 * @param parameters the parameters to be passed
+	 * @throws WorkerBusyException if this worker is already working
+	 */
+	public void doWork(Job job, Object parameters) throws BusyException {
+		if(busy)
+			throw new BusyException();
+		busy = true;
+		this.job = job;
+		this.parameters = parameters;
+		interrupt();
+	}
+	
+	@Override
+	public void run() {
+		while(alive) {
+			if(busy) {
+				Object result = job.run(parameters, shared);
+				master.notifyResult(result);
+				busy = false;
+			}
+			
+			try { sleep(10); }
+			catch(InterruptedException e) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)